### ISSUE TITLE
DNS over TLS (macOS Ventura)

### ISSUE THREAD

**Issue by soxrok2212** [pi-hole/pi-hole#4993]
### Versions

- Pi-hole:  v5.13
- AdminLTE:  v5.16
- FTL:  v5.18.2

### Platform

- OS and version:  Ubuntu 22

- Platform:  LXC

### Expected behavior

macOS respects pi-hole as optimal DNS server.

### Actual behavior / bug

It appears that macOS Ventura prefers DNS servers with TLS enabled regardless of order. e.g., if your DHCP server advertises, in this order, local pi-hole, 8.8.8.8, 1.1.1.1, macOS will choose 1.1.1.1 first as CloudFlare supports DNS over TLS.

### Steps to reproduce

1. Advertise multiple DNS servers in your DHCP settings. I do
- Local DNS
- Local DNS backup
- CloudFlare DNS
- Google DNS
2. Try to resolve a local IP via pihole. This won't work.
3. Remove CloudFlare server and resolve local IP again. This will now work.

The TL;DR is moving forward, macOS will prefer DNS over TLS regardless of order, so for users like myself that advertise some public DNS servers as a backup in case the local ones go down, local DNS will cease to work.

Also reported on reddit: https://www.reddit.com/r/pihole/comments/yd4nvk/comment/its9gha/?utm_source=share&utm_medium=web2x&context=3

I realize this isn't so much a pi-hole issue as a macOS issue, but I believe it will cease to work sooner-than-later if DNS over TLS isn't supported soon.

**Comment by PromoFaux** [comment]
>so for users like myself that advertise some public DNS servers as a backup in case the local ones go down,

Just a quick reply on this - your DHCP should only be advertising your pi-hole (if you're worried about redundancy, then spin up a second pi-hole device/container). There is no such thing (as far as I can recall) as an order of DNS servers, rather the client will generally send a request out to each, and then just use whichever responds first.

Problem goes away if you don't tell your clients to about Cloudflare...

**Comment by PromoFaux** [comment]
Actually this is also discussed in the linked reddit thread:

https://www.reddit.com/r/pihole/comments/yd4nvk/comment/itrnysp/?utm_source=reddit&utm_medium=web2x&context=3

https://discourse.pi-hole.net/t/why-should-pi-hole-be-my-only-dns-server/3376

**Comment by jfb-pihole** [comment]
"I realize this isn't so much a pi-hole issue as a macOS issue"

It's both, but more importantly it's a fundamental DNS issue.  Given multiple DNS servers from which to choose, clients are free to use any of them at any time.

I too run OS Ventura, and Pi-hole is the only DNS server advertised by DHCP.  All the DNS traffic goes through Pi-hole, as expected.
